Userland/Libraries/LibChess/Chess.cpp

```diff
@@ -221,42 +222,114 @@ String Move::to_algebraic() const
+            break;
+        case 'q':
+            cur_type = Type::Queen;
+            break;
+        case 'k':
+            cur_type = Type::King;
+            break;
+        case '/':
+            file = 0;
+            rank -= 1;
+            continue;
+        default:
+            return Error::from_string_literal("Invalid character in position string");
+        }
+        set_piece({ rank, file }, { cur_color, cur_type });
+        file += 1;
     }
 
-    // Fill black pawns.
-    for (int file = 0; file < 8; ++file) {
-        set_piece({ 6, file }, { Color::Black, Type::Pawn });
+    if (fen_color == "w") {
+        m_turn = Color::White;
+    } else if (fen_color == "b") {
+        m_turn = Color::Black;
+    } else {
+        return Error::from_string_literal("Bad character at color position");
+    }
+    bool white_king = false, white_queen = false,
+         black_king = false, black_queen = false;
+    for (auto c : fen_castling) {
+        switch (c) {
+        case 'K':
+            white_king = true;
+            break;
+        case 'Q':
+            white_queen = true;
+            break;
+        case 'k':
+            black_king = true;
+            break;
+        case 'q':
+            black_queen = true;
+            break;
+        case '-':
+            break;
+        default:
+            return Error::from_string_literal("Bad character at castling position");
+        }
     }
-
-    // Fill while pieces.
-    set_piece(Square("a1"), { Color::White, Type::Rook });
-    set_piece(Square("b1"), { Color::White, Type::Knight });
-    set_piece(Square("c1"), { Color::White, Type::Bishop });
-    set_piece(Square("d1"), { Color::White, Type::Queen });
-    set_piece(Square("e1"), { Color::White, Type::King });
-    set_piece(Square("f1"), { Color::White, Type::Bishop });
-    set_piece(Square("g1"), { Color::White, Type::Knight });
-    set_piece(Square("h1"), { Color::White, Type::Rook });
-
-    // Fill black pieces.
-    set_piece(Square("a8"), { Color::Black, Type::Rook });
-    set_piece(Square("b8"), { Color::Black, Type::Knight });
-    set_piece(Square("c8"), { Color::Black, Type::Bishop });
-    set_piece(Square("d8"), { Color::Black, Type::Queen });
-    set_piece(Square("e8"), { Color::Black, Type::King });
-    set_piece(Square("f8"), { Color::Black, Type::Bishop });
-    set_piece(Square("g8"), { Color::Black, Type::Knight });
-    set_piece(Square("h8"), { Color::Black, Type::Rook });
+    m_white_can_castle_kingside = white_king;
+    m_white_can_castle_queenside = white_queen;
+    m_black_can_castle_kingside = black_king;
+    m_black_can_castle_queenside = black_queen;
+    if (fen_enpassant_square.length() == 2)
+        m_enpassant_square = Chess::Square(fen_enpassant_square);
+    else if (fen_enpassant_square != "-")
+        return Error::from_string_literal("Bad character at en passant square position");
+    m_moves_since_capture = atoi(fen_halfmove.characters());
```

Comment:
We can use `StringView::to_int<T>` here, even with a `StringView`!

Reply:
Ah I missed that function. Updated!